validatedeps.js

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env node
```

Comment:
Checkdeps is pretty ungainly code and hard enough to follow as it is, probably better to keep them separate since they do separate things. We might want to rename them to make it clearer what they do, since they're both "checking" and "validating" dependencies.